modules/database-commons/src/main/java/org/testcontainers/ext/ScriptUtils.java

```diff
@@ -261,44 +258,14 @@ public static void executeSqlScript(Connection connection, String scriptPath, St
 
 			splitSqlScript(scriptPath, script, separator, commentPrefix, blockCommentStartDelimiter,
 				blockCommentEndDelimiter, statements);
-			int lineNumber = 0;
-			Statement stmt = connection.createStatement();
-			try {
-				for (String statement : statements) {
-					lineNumber++;
-					try {
-						stmt.execute(statement);
-						int rowsAffected = stmt.getUpdateCount();
-						if (LOGGER.isDebugEnabled()) {
-							LOGGER.debug(rowsAffected + " returned as updateCount for SQL: " + statement);
-						}
-					}
-					catch (SQLException ex) {
-						boolean dropStatement = statement.trim().toLowerCase().startsWith("drop");
-						if (continueOnError || (dropStatement && ignoreFailedDrops)) {
-							if (LOGGER.isDebugEnabled()) {
-								LOGGER.debug("Failed to execute SQL script statement at line " + lineNumber
-										+ " of resource " + scriptPath + ": " + statement, ex);
-							}
-						}
-						else {
-							throw new ScriptStatementFailedException(statement, lineNumber, scriptPath, ex);
-						}
-					}
-				}
-			}
-			finally {
-				try {
-					stmt.close();
-				}
-				catch (Throwable ex) {
-					LOGGER.debug("Could not close JDBC Statement", ex);
-				}
+
+			try (DatabaseDelegate closeableDelegate = databaseDelegate) {
```

Comment:
```
executeDatabaseScript(databaseDelegate, ...);
executeDatabaseScript(databaseDelegate, ...)
```

will not work because you close the delegate in this method. It should not perform such side effects

Reply:
To be fair, with current code it won't work either - because java.sql.Connection will be closed after first line. 
Okay, I created new boolean property in AbstractDatabaseDelegate (to avoid null based logic). Now first statement execution creates new connection. All subsequent statements will use existing connection. Closing the connection closes the connection and sets isConnectionStarted propery to false, if connection was started in the first place. All subsequent database delegate invokations will use the same logic.
>It should not perform such side effects

I disagree with you. Connection to the database should be closed after script invocation. As it is in the current diff or current code in master.